dockers/docker-teamd/start.sh

```diff
@@ -6,7 +6,12 @@ rm -rf $TEAMD_CONF_PATH
 mkdir -p $TEAMD_CONF_PATH
 
 SONIC_ASIC_TYPE=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type)
-MAC_ADDRESS=$(ip link show eth0 | grep ether | awk '{print $2}')
+
+if [ "$SONIC_ASIC_TYPE" == "mellanox" ]; then
+    MAC_ADDRESS=$(sonic-cfggen -d -v DEVICE_METADATA.localhost.mac)
+else
+    MAC_ADDRESS=$(ip link show eth0 | grep ether | awk '{print $2}')
```

Comment:
MAC_ADDRESS=$(</sys/class/net/eth0/address)

Reply:
the case under "else" getting MAC address from the "ip link show eth0" is the old code and used for other platforms
I can change it to what you suggested, but I am not able to test it on non-Mellanox boxes...